src/google-apis/auth-client.ts

```diff
@@ -94,21 +94,22 @@ export class RequestHandler extends common.Service {
    * @param {Configuration} config - an instance of the Configuration class
    * @param {Logger} logger - an instance of logger
    */
-  constructor(config: Configuration, logger: types.Logger) {
+  constructor(config: Configuration, logger: Logger) {
     const pid = config.getProjectId();
     // If an API key is provided, do not try to authenticate.
     const tryAuthenticate = !config.getKey();
+    const serviceOptions: ServiceOptions = Object.assign(config, {
```

Comment:
@DominicKramer it looks like this may have been the source of a buggy bug.  Did I interpret your intent here right?